crates/scroll/alloy/consensus/src/receipt/receipt.rs

```diff
@@ -0,0 +1,256 @@
+//! Transaction receipt types for Scroll.
+
+use alloy_consensus::{
+    Eip658Value, Receipt, ReceiptWithBloom, RlpDecodableReceipt, RlpEncodableReceipt, TxReceipt,
+};
+use alloy_primitives::{Bloom, Log, U256};
+use alloy_rlp::{Buf, BufMut, Decodable, Encodable, Header};
+
+/// Receipt containing result of transaction execution.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "camelCase"))]
+pub struct ScrollTransactionReceipt<T = Log> {
+    /// The inner receipt type.
+    #[cfg_attr(feature = "serde", serde(flatten))]
+    pub inner: Receipt<T>,
+    /// L1 fee for Scroll transactions.
+    pub l1_fee: U256,
```

Comment:
I know we discussed about using `Option` here. I think the only place where the node could lose some information due to incorrect decoding of the receipt would be if the node requests receipts via the `GetReceipts` message on the eth-wire. Unless we plan on adding code in order to recover from such a loss of info, I would argue that adding the `Option` doesn't bring anything.

Reply:
What are the implications of not having this field returned via the `GetReceipts` request on eth-wire?

Reply:
I don't think the `GetReceipts` response is ever handled in Reth:
- [poll implementation](https://github.com/scroll-tech/reth/blob/2f46fe6d48407044ad58a3ae37f59380b57044aa/crates/net/network/src/state.rs#L483) for the `NetworkState` handles received responses.
- The `GetReceipts` variant is [ignored](https://github.com/scroll-tech/reth/blob/2f46fe6d48407044ad58a3ae37f59380b57044aa/crates/net/network/src/state.rs#L412).